### PR TITLE
M5 dataset

### PR DESCRIPTION
**Pull request by FedericoGarza**
- M5 dataset wrangling﻿ based on @jose-moralez's work (https://www.kaggle.com/lemuz90/m5-preprocess)
- Added M5Evaluation class and test sota results.

### CODE REVIEW

File: nixtlats/data/datasets/m5.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: nbs/data_datasets__m5.ipynb (unless otherwise specified).
+
+__all__ = ['M5']
+
+# Cell
+import os
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple, Union
+
+from numba import njit
```
Comment (jose-moralez) [inline nixtlats/data/datasets/m5.py:10]:
```suggestion
```

File: nixtlats/data/datasets/m5.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        @njit
+        def first_nz_mask(x):
+            """Return a boolean mask where the True starts at the first non-zero value."""
+            mask = np.full(x.size, True)
+            for idx, value in enumerate(x):
+                if value == 0:
+                    mask[idx] = False
+                else:
+                    break
+            return mask
```
Comment (jose-moralez) [inline nixtlats/data/datasets/m5.py:122]:
```suggestion
        def first_nz_mask(values, index):
            """Return a boolean mask where the True starts at the first non-zero value."""
            mask = np.full(values.size, True)
            for idx, value in enumerate(values):
                if value == 0:
                    mask[idx] = False
                else:
                    break
            return mask
```

File: nixtlats/data/datasets/m5.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+        long = long.sort_values(['id', 'date'], ignore_index=True)
+        keep_mask = long.groupby('id')['y'].transform(lambda x: first_nz_mask(x.values))
+        long = long[keep_mask]
```
Comment (jose-moralez) [inline nixtlats/data/datasets/m5.py:126]:
```suggestion
        keep_mask = long.groupby('id')['y'].transform(first_nz_mask, engine='numba')
        long = long[keep_mask.astype(bool)]
```

File: nixtlats/data/datasets/m5.py (hunk 1 above)
Comment (FedericoGarza) [reply nixtlats/data/datasets/m5.py]: Solved in https://github.com/Nixtla/nixtlats/pull/63/commits/93a8f5490f54e7387d4964b997d92b5763ab545d

File: nixtlats/data/datasets/m5.py (hunk 2 above)
Comment (FedericoGarza) [reply nixtlats/data/datasets/m5.py]: Solved in https://github.com/Nixtla/nixtlats/pull/63/commits/93a8f5490f54e7387d4964b997d92b5763ab545d

File: nixtlats/data/datasets/m5.py (hunk 3 above)
Comment (FedericoGarza) [reply nixtlats/data/datasets/m5.py]: Solved in https://github.com/Nixtla/nixtlats/pull/63/commits/93a8f5490f54e7387d4964b997d92b5763ab545d

**Review by kdgutier — APPROVED** [review]

**Pull request merged**